from urllib2 import Request, urlopen, URLError, HTTPError
from subprocess import call
import sys
import json
import HTMLParser

workingdir = './'

#send request to lolcounter

req = Request('https://prod.api.pvp.net/api/lol/static-data/na/v1.2/champion?api_key=1307d9e7-3353-4720-bda4-13baf7ba200b')
try:
    response = urlopen(req)
except HTTPError as e:
	f = open(workingdir+"error.json", 'w')
	f.write('{"ServerError":"The server couldn\'t fulfill the request.  Error code '+str(e.code)+'"}')
	f.close();
	sys.exit(1)
except URLError as e:
	f = open(workingdir+"error.json", 'w')
	f.write('{"ServerError":"We failed to reach a server.  Reason: '+str(e.reason)+'"}')
	f.close();
	sys.exit(1)
    #print 'We failed to reach a server.'
    #print 'Reason: ', e.reason
else:
	returnString = response.read()
	json = json.loads(returnString)
	for i in json["data"]:
		print json["data"][i]["name"]
		call(["python", workingdir+"get_counter_info.py", json["data"][i]["name"]])
	

sys.exit(0)
